agent/engine/dockerstate/docker_task_engine_state.go

```diff
@@ -96,6 +98,18 @@ func newDockerTaskEngineState() *DockerTaskEngineState {
 	}
 }
 
+// Reset resets all the states
+func (state *DockerTaskEngineState) Reset() {
```

Comment:
Can you modify this so that you extract all the `state.xxx=make()` bits into a new method, and invoke it here and from `newDockerTaskEngineState()` so that in future if we modify this struct to persist more state, there's no cognitive load of updating `Reset` method as well?